## Aufgabe 2

Für die zweite Aufgabe wurde die Hg-Spektrallampe ausgetauscht für eine Na-Spektrallampe, außerhalb davon blieb der Aufbau gleich. Es wurde wieder damit begonnen, den Korrekturwinkel $\phi_0$ zu bestimmen. Für diesen haben wir $\phi_0 = 359,5928°$ gemessen, oder auch $\phi_0 = -0,4072°$. Genau wie in Aufgabe 1 wurde auch das Reflexionsgitter so eingestellt, das bei ca. 60° das Maximum 0. Ordnung zu sehen ist. Genauer liegt dies bei $Max_0 = 60,1078°$.

Bei der Na-Spektrallampe kann man nicht nur einzelen Maxima sehen sondern durch die Spin-Bahn-Kopplung bei Natrium kann man 4 verschiedene Dubletts erkennen in den Farben Rot, Orange, Grün und Blau. Die Maxima erster Ordnung von allen Dubletts wurden gemessen und folgende Werte haben wir abgelesen:

| Ordnung  | Grad [°] | Minuten ['] | Sekunden [''] | Gesamt [°]    |
|----------|---------:|------------:|---------------:|------------:|
| Blau 1   |       87 |          16 |            28 $\pm 2$  | 87,27444444 $\pm 2''$|
| Blau 2   |       87 |          17 |            32 $\pm 2$  | 87,29222222 $\pm 2''$|
| Grün 1   |       90 |          21 |            22  $\pm 2$ | 90,35611111 $\pm 2''$|
| Grün 2   |       90 |          22 |            36  $\pm 2$ | 90,37666667 $\pm 2''$|
| Orange 1 |       91 |          14 |            18  $\pm 2$ | 91,23833333 $\pm 2''$|
| Orange 2 |       91 |          15 |            28  $\pm 2$ | 91,25777778 $\pm 2''$|
| Rot 1    |       92 |          21 |            28  $\pm 2$ | 92,35777778 $\pm 2''$|
| Rot 2    |       92 |          22 |            48  $\pm 2$ | 92,38000000 $\pm 2''$|

Da bei $Max_0$ gilt $\alpha = \beta$, können wir wieder mit der Formel $\alpha = \frac{1}{2}(180° - (Max_0 - \phi_0))$ berechnen. Daraus folgt $\alpha = \frac{1}{2}(180° - (60,1078° + 0,4072))= 59,7425°$

Für die Berechnung der Wellenlänge nutzen wir die Formel: 
$$m \cdot \lambda = g \cdot (sin(\alpha) - sin(\beta))$$
wobei wir immer nur das erste Maximum betrachten, weshalb immer m=1 gilt. Den Winkel für $\beta$ können wir berechnen mit $\beta = 180° - (\alpha + (\phi - \phi_0))$.

Für die Gitterkonstante g nutzen wir den aus Aufgabe 1 berechneten Wert: $g = 1.544 \cdot 10 ^{-6}m$

In [2]:
import math
import numpy as np

# Winkelwerte in Grad
blau_1 = 87.27444444
blau_2 = 87.29222222
gruen_1 = 90.35611111
gruen_2 = 90.37666667
orange_1 = 91.23833333
orange_2 = 91.25777778
rot_1 = 92.35777778
rot_2 = 92.38

korrekturwinkel = 0.4072
alpha = 59.7425
d = 1.544e-6  # Gitterkonstante in Metern

def get_wavelength(phi):
    beta = 180 - (alpha + (phi - korrekturwinkel))  # Richtige Formel!
    wavelength = d * (np.sin(alpha * np.pi/180) - (np.sin(beta * np.pi/180))) * 1e9
    return wavelength

def main():
    colors = {
        "blau_1": blau_1,
        "blau_2": blau_2,
        "gruen_1": gruen_1,
        "gruen_2": gruen_2,
        "orange_1": orange_1,
        "orange_2": orange_2,
        "rot_1": rot_1,
        "rot_2": rot_2
    }

    for name, angle in colors.items():
        wl = get_wavelength(angle)
        print(f"{name}: {wl:.3f} nm")

if __name__ == "__main__":
    main()


blau_1: 483.937 nm
blau_2: 484.337 nm
gruen_1: 554.470 nm
gruen_2: 554.948 nm
orange_1: 575.086 nm
orange_2: 575.542 nm
rot_1: 601.504 nm
rot_2: 602.031 nm


Somit haben wir für unsere verschiedenen Dubletts die folgenden berechneten Wellenlängen:

| Ordnung  | Gesamt [°]  | Wellenlänge [nm]
|----------|------------:|------------:|
| Blau 1   | 87,27444444 | 483.937 |
| Blau 2   | 87,29222222 | 484.337 |
| Grün 1   | 90,35611111 | 554.470 |
| Grün 2   | 90,37666667 | 554.948 |
| Orange 1 | 91,23833333 | 575.086 |
| Orange 2 | 91,25777778 | 575.542 |
| Rot 1    | 92,35777778 | 601.504 |
| Rot 2    | 92,38000000 | 602.031 |

## Vergleich mit Literaturwerten und Diskussion


Wir vergleichen nun unsere Werte mit den Literaturwerten für das Natrium Spektrum, dafür nutzen wir die NIST Datenbank "https://physics.nist.gov/PhysRefData/Handbook/Tables/sodiumtable3.htm". In der Tabelle hinzugefügt sind auch die Abweichnungen zu unseren berechneten Wellenlängen: 
| Farbe (subjektiv)   | Wellenlänge 1 [nm] | Wellenlänge 2 [nm] |  Δλ [nm] | Abweichung Wellenlänge 1 | Abweichung Wellenlänge 2 |
| :------------------ | -----------------: | -----------------: | -------: | ------------------------ | ------------------------ |
| **Blau**            |         466.481 nm |         466.856 nm | 0.375 nm | 3,6%.                    |        3,61%             |
| **Grün**            |         497.854 nm |         498.281 nm | 0.427 nm |    10,21%                |         10,21%           |
| **Orange**          |         568.263 nm |         568.819 nm | 0.556 nm |       1,19%.             | 1,17%                    |
| **Gelb (D-Linien)** |         588.995 nm |         589.592 nm | 0.597 nm |           2,08%          |    2,07%                 |

Die gemessenen Wellenlängen der Natrium-Dubletts liegen im sichtbaren Bereich und zeigen den erwarteten Trend steigender Wellenlänge mit zunehmender Ordnung. Im Vergleich zu den Literaturwerten aus der NIST-Datenbank betragen die Abweichungen meist unter 3,5 %, mit Ausnahme des grünen Dubletts. Die Unterschiede lassen sich auf Justageungenauigkeiten, Messfehler bei den Winkeln sowie eine leicht abweichende Gitterkonstante zurückführen. Da die Farbunterscheidung in der ersten Ordnung teilweise schwierig war und sich in der Nähe bereits Linien höherer Ordnung befanden, ist es möglich, dass beim grünen Dublett versehentlich eine benachbarte Linie abgelesen wurde – was die deutlich größere Abweichung von rund 10 % besser erklärt. Abgesehen vom grünen sind aber alle Abweichungen im typischen Bereich von Messunsicherheiten.
